src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,95 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imageURL, setImageURL] = useState('');
+  const [imdbURL, setImdbURL] = useState('');
+  const [imdbID, setImdbID] = useState('');
+
+  const refreshForm = () => {
+    setTitle('');
+    setDescription('');
+    setImageURL('');
+    setImdbURL('');
+    setImdbID('');
+  };
+
+  const disabledButton: boolean =
+    !title.trim() || !imageURL.trim() || !imdbURL.trim() || !imdbID.trim();
+
+  const buttonHandler = (event: React.FormEvent) => {
```

Comment:
let's follow common naming convention for eventHandlers
please read this [article](https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364) for more details

```suggestion
  const handleFormSubmit = (event: React.FormEvent) => {
```